extensions/MozChangeField/lib/Post/TypePriSevEditbugs.pm

```diff
@@ -0,0 +1,28 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::MozChangeField::Post::TypePriSevEditbugs;
+
+use 5.10.1;
+use Moo;
+
+use Bugzilla::Constants;
+
+sub evaluate_change {
+  my ($self, $args) = @_;
+  my $changes  = $args->{changes};
+  my $editbugs = $args->{editbugs};
+
+  # If changing the bug_type, severity, or priority, editbugs is required.
+  foreach my $field (qw(bug_type bug_severity priority)) {
+    if (exists $changes->{$field} && !$editbugs) {
+      ThrowUserError('mozchangefield_field_needs_editbugs', {field => $field});
```

Comment:
This should return the error and reasons as all other checkers instead of raising the error directly.

Reply:
Those type of return values only apply to Pre instances (bug_check_can_change_field) and Post instances are for after changes have been submitted. So Post instances only full stop using Throw{Code|User}Error() or they modify the data in some way.